packages/rest/tests/webhook.test.ts

```diff
@@ -0,0 +1,108 @@
+import type { WebhookData } from './utils/webhook'
+import type { Agent, CredentialStateChangedEvent, ProofStateChangedEvent } from '@aries-framework/core'
+import type { Server } from 'http'
+
+import {
+  ProofEventTypes,
+  ProofState,
+  ProofRecord,
+  CredentialState,
+  CredentialRecord,
+  CredentialEventTypes,
+} from '@aries-framework/core'
+import { EventEmitter } from '@aries-framework/core/build/agent/EventEmitter'
+
+import { setupServer } from '../src/server'
+
+import { getTestAgent } from './utils/helpers'
+import { webhookListener } from './utils/webhook'
+
+describe('WebhookTest', () => {
+  let agent: Agent
+  let server: Server
+  const webhooks: WebhookData[] = []
+
+  beforeEach(async () => {
+    agent = await getTestAgent('Rest Webhook Test Bob', 3003)
+    server = await webhookListener(3000, webhooks)
+    await setupServer(agent, { webhookUrl: 'http://localhost:3000', port: 3004 })
+  })
+
+  afterEach(async () => {
+    await agent.shutdown()
+    await agent.wallet.delete()
+    server.close()
+  })
+
+  test('should return a webhook event when connection state changed', async () => {
+    const { connectionRecord } = await agent.connections.createConnection()
+
+    const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
```

Comment:
Maybe we can define this once in the `utils/helpers` file and reuse for all tests? Declaring it multiple times seems uncessary